src/Middleware/HttpLogging/src/HttpLoggingServicesExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.HttpLogging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for the HttpLogging middleware.
+    /// </summary>
+    public static class HttpLoggingServicesExtensions
+    {
+        /// <summary>
+        /// Adds HTTP Logging services.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> for adding services.</param>
+        /// <returns></returns>
+        public static IServiceCollection AddHttpLogging(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.TryAddSingleton<HttpLoggingOptions>();
```

Comment:
Is this required? I think config will just create whatever type you need with `IOptions<T>` without registering it in DI.

Reply:
Yeah I don't need this.